pkg/asset/manifests/azure/cloudproviderconfig.go

```diff
@@ -23,7 +26,7 @@ func (params CloudProviderConfig) JSON() (string, error) {
 	resourceGroupName := params.ResourcePrefix + "-rg"
 	config := config{
 		authConfig: authConfig{
-			Cloud:                       "AzurePublicCloud",
+			Cloud:                       string(params.CloudName),
```

Comment:
Hm I dislike casting or type conversion in general but I don't think there's anyway around this.

Reply:
I could add a function like the following.
```
// Environment returns name that Azure uses for the cloud environment.
// See https://github.com/Azure/go-autorest/blob/ec5f4903f77ed9927ac95b19ab8e44ada64c1356/autorest/azure/environments.go#L13
func (e CloudEnvironment) Name() string {
	return string(e)
}
```
That would push the casting to a location closer to where the enum is defined. I don't know of any precedence for this approach, though.
```suggestion
			Cloud:                       params.CloudName.Name(),
```